api/src/test_aid/systest_base.py

```diff
@@ -14,73 +16,87 @@
 from selenium.webdriver.common.by import By
 
 from service.config import get_mysql_config
-from service.db import create_mysql_engine, db_session
-from shop.stripe_constants import Type
-from test_aid.api import ApiFactory
+from service.db import create_mysql_engine, db_session_factory, db_session
+from shop.stripe_constants import EventType, set_stripe_key
+from test_aid.api import ApiFactory, ApiResponse
 from test_aid.db import DbFactory
 from test_aid.obj import DEFAULT_PASSWORD
 from test_aid.systest_config import HOST_FRONTEND, HOST_PUBLIC, HOST_BACKEND, \
     SELENIUM_BASE_TIMEOUT, SLEEP, WEBDRIVER_TYPE, TEST_SERVICE_TOKEN, SELENIUM_SCREENSHOT_DIR, KEEP_BROWSER, \
     STRIPE_PUBLIC_KEY
 from test_aid.test_base import TestBase, ShopTestMixin
 
-stripe.api_key = STRIPE_PUBLIC_KEY
-
 VALID_NON_3DS_CARD_NO = "378282246310005"
 VALID_3DS_CARD_NO = "4242424242424242"
 EXPIRED_3DS_CARD_NO = "4000000000000069"
-
+DECLINE_AFTER_ATTACHING_CARD = "4000000000000341"
 EXPIRED_CVC_ZIP = "4242424242424"
 
-
 logger = getLogger('makeradmin')
 
 
 class SystestBase(TestBase):
     """ Base class for systest with config available. """
-    
+    db: DbFactory
+    admin_url: str
+    public_url: str
+    api_url: str
+
     @classmethod
-    def setUpClass(self):
+    def setUpClass(self) -> None:
         super().setUpClass()
+
+        # Use the public key for these tests. TODO: Why not the private one?
+        set_stripe_key(False)
```

Comment:
Pass boolean argument by name

```suggestion
        set_stripe_key(private=False)
```